core/model/modx/processors/resource/trash/purge.class.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * Empties the recycle bin.
+ *
+ * @return boolean
+ *
+ * @package    modx
+ * @subpackage processors.resource
+ */
+class modResourceTrashPurgeProcessor extends modProcessor {
+
+    /** @var modResource $resource */
+    public $resource;
+
+    /**
+     * @var int The id of the resource to be deleted.
+     */
+    public $id;
+
+
+    public function checkPermissions() {
+        return $this->modx->hasPermission( 'purge_deleted' );
+    }
+
+    public function getLanguageTopics() {
+        return array( 'resource' );
+    }
+
+    /**
+     * @return bool|null|string
+     */
+    public function initialize() {
+        $this->id = $this->getProperty( 'id', false );
+        if ( empty( $this->id ) ) return $this->modx->lexicon( 'resource_err_ns' );
+
+        $this->resource = $this->modx->getObject( 'modResource', $this->id );
+        if ( empty( $this->resource ) ) return $this->modx->lexicon( 'resource_err_nfs', array( 'id' => $this->id ) );
```

Comment:
If you pass a comma separated list here, this should not work, preventing "batch purging"